File: src/main/scala-graphql/io/github/graphql/scala/model/GpgSignatureTO.scala
package io.github.graphql.scala.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import java.util.Objects
import scala.collection.JavaConverters._
import GitSignatureStateTO._

@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-07T19:49:36+0800"
)
class GpgSignatureTO(
    @javax.validation.constraints.NotNull
    override val email: String,
    override val isValid: Boolean,
    keyId: String,
    @javax.validation.constraints.NotNull
    override val payload: String,
    @javax.validation.constraints.NotNull
    override val signature: String,
    override val signer: UserTO,
    @javax.validation.constraints.NotNull
    override val state: GitSignatureStateTO,
    override val wasSignedByGitHub: Boolean
) extends GitSignatureTO {

    override def toString(): String = {
        Seq(
            if (email != null) "email: " + GraphQLRequestSerializer.getEntry(email) else "",
            "isValid: " + GraphQLRequestSerializer.getEntry(isValid),
            if (keyId != null) "keyId: " + GraphQLRequestSerializer.getEntry(keyId) else "",
            if (payload != null) "payload: " + GraphQLRequestSerializer.getEntry(payload) else "",
            if (signature != null) "signature: " + GraphQLRequestSerializer.getEntry(signature) else "",
            if (signer != null) "signer: " + GraphQLRequestSerializer.getEntry(signer) else "",
            if (state != null) "state: " + GraphQLRequestSerializer.getEntry(state) else "",
            "wasSignedByGitHub: " + GraphQLRequestSerializer.getEntry(wasSignedByGitHub)
        ).filter(_ != "").mkString("{", ",", "}")
    }

    override def equals(obj: Any): Boolean = {
        if (this == obj) {
            return true
        }
        if (obj == null || getClass != obj.getClass) {
            return false
        }
        val that = obj.asInstanceOf[GpgSignatureTO]
        Objects.equals(email, that.email) &&
        Objects.equals(isValid, that.isValid) &&
        Objects.equals(keyId, that.keyId) &&
        Objects.equals(payload, that.payload) &&
        Objects.equals(signature, that.signature) &&
        Objects.equals(signer, that.signer) &&
        Objects.equals(state, that.state) &&
        Objects.equals(wasSignedByGitHub, that.wasSignedByGitHub)
    }

    override def hashCode(): Int = {
        Objects.hash(email, isValid, keyId, payload, signature, signer, state, wasSignedByGitHub)
    }
}

object GpgSignatureTO {

    def builder(): GpgSignatureTO.Builder = new Builder()

    class Builder {

        private var email: String = _
        private var isValid: Boolean = _
        private var keyId: String = _
        private var payload: String = _
        private var signature: String = _
        private var signer: UserTO = _
        private var state: GitSignatureStateTO = _
        private var wasSignedByGitHub: Boolean = _

        def setEmail(email: String): Builder = {
            this.email = email
            this
        }

        def setIsValid(isValid: Boolean): Builder = {
            this.isValid = isValid
            this
        }

        def setKeyId(keyId: String): Builder = {
            this.keyId = keyId
            this
        }

        def setPayload(payload: String): Builder = {
            this.payload = payload
            this
        }

        def setSignature(signature: String): Builder = {
            this.signature = signature
            this
        }

        def setSigner(signer: UserTO): Builder = {
            this.signer = signer
            this
        }

        def setState(state: GitSignatureStateTO): Builder = {
            this.state = state
            this
        }

        def setWasSignedByGitHub(wasSignedByGitHub: Boolean): Builder = {
            this.wasSignedByGitHub = wasSignedByGitHub
            this
        }

        def build(): GpgSignatureTO = new GpgSignatureTO(email, isValid, keyId, payload, signature, signer, state, wasSignedByGitHub)

    }
}

File: src/main/java-graphql/io/github/graphql/java/model/GpgSignatureTO.java
package io.github.graphql.java.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer;
import java.util.Objects;
import java.util.StringJoiner;

@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-07T19:46:14+0800"
)
public class GpgSignatureTO implements java.io.Serializable, GitSignatureTO {

    @javax.validation.constraints.NotNull
    private String email;
    private boolean isValid;
    private String keyId;
    @javax.validation.constraints.NotNull
    private String payload;
    @javax.validation.constraints.NotNull
    private String signature;
    private UserTO signer;
    @javax.validation.constraints.NotNull
    private GitSignatureStateTO state;
    private boolean wasSignedByGitHub;

    public GpgSignatureTO() {
    }

    public GpgSignatureTO(String email, boolean isValid, String keyId, String payload, String signature, UserTO signer, GitSignatureStateTO state, boolean wasSignedByGitHub) {
        this.email = email;
        this.isValid = isValid;
        this.keyId = keyId;
        this.payload = payload;
        this.signature = signature;
        this.signer = signer;
        this.state = state;
        this.wasSignedByGitHub = wasSignedByGitHub;
    }

    public String getEmail() {
        return email;
    }

    public boolean getIsValid() {
        return isValid;
    }

    public String getKeyId() {
        return keyId;
    }

    public String getPayload() {
        return payload;
    }

    public String getSignature() {
        return signature;
    }

    public UserTO getSigner() {
        return signer;
    }

    public GitSignatureStateTO getState() {
        return state;
    }

    public boolean getWasSignedByGitHub() {
        return wasSignedByGitHub;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null || getClass() != obj.getClass()) {
            return false;
        }
        final GpgSignatureTO that = (GpgSignatureTO) obj;
        return Objects.equals(email, that.email)
            && Objects.equals(isValid, that.isValid)
            && Objects.equals(keyId, that.keyId)
            && Objects.equals(payload, that.payload)
            && Objects.equals(signature, that.signature)
            && Objects.equals(signer, that.signer)
            && Objects.equals(state, that.state)
            && Objects.equals(wasSignedByGitHub, that.wasSignedByGitHub);
    }

    @Override
    public int hashCode() {
        return Objects.hash(email, isValid, keyId, payload, signature, signer, state, wasSignedByGitHub);
    }

    @Override
    public String toString() {
        StringJoiner joiner = new StringJoiner(", ", "{ ", " }");
        if (email != null) {
            joiner.add("email: " + GraphQLRequestSerializer.getEntry(email));
        }
        joiner.add("isValid: " + GraphQLRequestSerializer.getEntry(isValid));
        if (keyId != null) {
            joiner.add("keyId: " + GraphQLRequestSerializer.getEntry(keyId));
        }
        if (payload != null) {
            joiner.add("payload: " + GraphQLRequestSerializer.getEntry(payload));
        }
        if (signature != null) {
            joiner.add("signature: " + GraphQLRequestSerializer.getEntry(signature));
        }
        if (signer != null) {
            joiner.add("signer: " + GraphQLRequestSerializer.getEntry(signer));
        }
        if (state != null) {
            joiner.add("state: " + GraphQLRequestSerializer.getEntry(state));
        }
        joiner.add("wasSignedByGitHub: " + GraphQLRequestSerializer.getEntry(wasSignedByGitHub));
        return joiner.toString();
    }

    public static GpgSignatureTO.Builder builder() {
        return new GpgSignatureTO.Builder();
    }

    public static class Builder {

        private String email;
        private boolean isValid;
        private String keyId;
        private String payload;
        private String signature;
        private UserTO signer;
        private GitSignatureStateTO state;
        private boolean wasSignedByGitHub;

        public Builder() {
        }

        public Builder setEmail(String email) {
            this.email = email;
            return this;
        }

        public Builder setIsValid(boolean isValid) {
            this.isValid = isValid;
            return this;
        }

        public Builder setKeyId(String keyId) {
            this.keyId = keyId;
            return this;
        }

        public Builder setPayload(String payload) {
            this.payload = payload;
            return this;
        }

        public Builder setSignature(String signature) {
            this.signature = signature;
            return this;
        }

        public Builder setSigner(UserTO signer) {
            this.signer = signer;
            return this;
        }

        public Builder setState(GitSignatureStateTO state) {
            this.state = state;
            return this;
        }

        public Builder setWasSignedByGitHub(boolean wasSignedByGitHub) {
            this.wasSignedByGitHub = wasSignedByGitHub;
            return this;
        }


        public GpgSignatureTO build() {
            return new GpgSignatureTO(email, isValid, keyId, payload, signature, signer, state, wasSignedByGitHub);
        }

    }
}
